src/pytest_workflow/content_tests.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (C) 2018 Leiden University Medical Center
+# This file is part of pytest-workflow
+#
+# pytest-workflow is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# pytest-workflow is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with pytest-workflow.  If not, see <https://www.gnu.org/licenses/
+
+"""This contains all the classes and methods for content testing of files
+and logs."""
+
+from pathlib import Path
+from typing import Dict, Iterable, List
+
+import pytest
+
+from .schema import ContentTest
+
+
+def check_content(strings: List[str],
+                  text_lines: Iterable[str]) -> Dict[str, bool]:
+    """
+    Checks whether any of the strings is present in the text lines
+    This function is optimized for big texts, it stops searching when
+    everything is found.
+    :param strings: A list of strings for which the present is checked
+    :param text_lines: The lines of text that need to be searched.
+    :return:
+    """
+    # Make a copy of the list here to prevent aliasing.
+    not_found_strings = list(strings)
+    # By default all strings are not found.
+    found_dictionary = {key: False for key in not_found_strings}
```

Comment:
I think a dictionary here is redundant. A set would serve the same purpose.
```suggestion
    found_items = set()
```